docker/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Dockerfile to build the matrixdotorg/sygnal docker images.
+#
+# To build the image, run `docker build` command from the root of the
+# sygnal repository:
+#
+#    docker build -f docker/Dockerfile .
+#
+
+###
+### Stage 0: builder
+###
+FROM python:3.7-slim as builder
+
+# build things which have slow build steps, before we copy sygnal, so that
+# the layer can be cached.
+#
+# (we really just care about caching a wheel here, as the "pip install" below
+# will install them again.)
+
+RUN pip install --prefix="/install" --no-warn-script-location cryptography
+
```

Comment:
There are a few more build dependencies now, mostly because of PostgreSQL support

```suggestion

# install required build dependencies

RUN apt-get update && apt-get install -y gcc git libpq-dev

```